examples/flutter_gallery/lib/demo/cupertino/cupertino_navigation_demo.dart

```diff
@@ -0,0 +1,740 @@
+    const Color.fromARGB(255, 0, 122, 255),
+    const Color.fromARGB(255, 88, 86, 214),
+    const Color.fromARGB(255, 255, 45, 85),
+  ];
+
+  static const List<String> coolColorNames = const <String>[
+    'Sarcoline', 'Coquelicot', 'Smaragdine', 'Mikado', 'Glaucous', 'Wenge',
+    'Fulvous', 'Xanadu', 'Falu', 'Eburnean', 'Amaranth', 'Australien',
+    'Banan', 'Falu', 'Gingerline', 'Incarnadine', 'Labrabor', 'Nattier',
+    'Pervenche', 'Sinoper', 'Verditer', 'Watchet', 'Zaffre',
+  ];
+
+  Tab1RowItemState() :
+      color = coolColors[new math.Random().nextInt(coolColors.length)],
+      colorName = coolColorNames[new math.Random().nextInt(coolColorNames.length)];
+
+  final Color color;
+  final String colorName;
+
+  bool added = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget row = new GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      onTap: () {
+        Navigator.of(context).push(new CupertinoPageRoute<Null>(
+          builder: (BuildContext context) => new Tab1ItemPage(
+            color: color,
+            colorName: colorName,
+            index: widget.index,
+          ),
+        ));
+      },
+      child: new Padding(
+        padding: const EdgeInsets.only(left: 16.0, top: 8.0, bottom: 8.0, right: 8.0),
+        child: new Row(
+          children: <Widget>[
+            new Container(
+              height: 60.0,
+              width: 60.0,
+              decoration: new BoxDecoration(
+                color: color,
+                borderRadius: new BorderRadius.circular(8.0),
+              ),
+            ),
+            new Expanded(
+              child: new Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 12.0),
+                child: new Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: <Widget>[
+                    new Text(colorName),
+                    const Padding(padding: const EdgeInsets.only(top: 8.0)),
+                    const Text(
+                      'Buy this cool color',
+                      style: const TextStyle(
+                        color: const Color(0xFF8E8E93),
+                        fontSize: 13.0,
+                        fontWeight: FontWeight.w300,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+            new CupertinoButton(
+              padding: EdgeInsets.zero,
+              child: new Icon(
+                added ? CupertinoIcons.minus_circled : CupertinoIcons.plus_circled,
+                color: CupertinoColors.activeBlue
+              ),
+              onPressed: () {
+                setState(() { added = !added; });
+              },
+            ),
+            new CupertinoButton(
+              padding: EdgeInsets.zero,
+              child: const Icon(CupertinoIcons.share, color: CupertinoColors.activeBlue),
+              onPressed: () {},
```

Comment:
nit: empty code blocks have a space inside them, to distinguish them from empty maps

Do we want this button to do something, like show a dialog?

Reply:
Done

I thought about actually bringing in the share plugin and actually share something but thought it might be a bit too heavy